pkg/sync/sync_context.go

```diff
@@ -252,6 +253,7 @@ type syncContext struct {
 	lock sync.Mutex
 
 	createNamespace bool
+	appLabelKey     string
```

Comment:
App label is Argo CD specific implementation detail. Instead of hardcoding app label handling here, I suggest providing the function pointer that applies required modification on namespace object